engine/compiler/compiler.go

```diff
@@ -802,12 +807,13 @@ func packNetrcSecrets(spec *engine.Spec, netrc *drone.Netrc) bool {
 }
 
 // setNetrcSecretsToStep is a helper function that sets netrc secrets to a engine.Step
-func setNetrcSecretsToStep(step *engine.Step, spec *engine.Spec) {
+func setNetrcSecretsToStep(step *engine.Step, spec *engine.Spec, secretMap map[string]*engine.Secret) {
 	envVars := []string{envNetrcMachine, envNetrcUsername, envNetrcPassword, envNetrcFile}
 	for _, envVar := range envVars {
 		if v, ok := spec.Secrets[envVar]; ok {
 			step.Secrets = append(step.Secrets, &engine.SecretVar{Name: v.Name, Env: v.Name})
 			step.SpecSecrets = append(step.SpecSecrets, v)
+			secretMap[v.Name] = v
```

Comment:
are we using the value? If you are only checking for existence of a name, simply use struct{} as the value is the map.